### PR TITLE
Upgrade: Update pdf.js to v2.0.943 and closure compiler to v20190121

### PR DESCRIPTION
**Pull request by jstoffan**
**Summary**
This changeset is partially based on and supersedes #810. I've migrated our calls to pdf.js to match the associated API changes from 1.x -> 2.x. However, there is no official migration guide, so further manual testing on document variations will be required.

**Known Issues**
- [ ] **Blocker**: Find controls do not work properly and do not highlight search results
- [x] **Blocker**: Missing glyphs and console warning for documents with invalid fonts: `Warning: Failed to load font "g_d0_f2": SyntaxError: Invalid font data in ArrayBuffer`
- [x] **Minor**: Console error on preview load for documents only in DocumentViewer.pagesinitHandler: `Uncaught TypeError: Cannot read property 'div' of undefined`

**Features Tested**
- Documents with digital signatures
- Documents with single and multiple pages
- Presentations with single and multiple pages
- Fullscreen mode via on-screen controls
- Page-up/down via on-screen controls and scroll wheel
- Text layer highlighting via click-drag and double-click
- Zoom-in/out via on-screen controls

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @jstoffan has signed the CLA. Thanks for the pull request!

File: src/lib/util.js [hunk 1]
```diff
@@ -508,28 +521,33 @@ export function loadScripts(urls, disableAMD = false) {
     }
 
     urls.forEach((url) => {
-        if (!head.querySelector(`script[src="${url}"]`)) {
+        if (!promiseMap[url]) {
```
Comment (DanDeMicco) [inline src/lib/util.js]: Would this function be cleaner as a map?

File: src/lib/util.js [hunk 2]
```diff
@@ -13,6 +13,8 @@ const CLIENT_NAME = __NAME__;
 export const CLIENT_VERSION = __VERSION__;
 /* eslint-enable no-undef */
 
+const promiseMap = {};
```
Comment (DanDeMicco) [inline src/lib/util.js]: Can you use an actual `Map`? This way you get built in methods such as clear for free (which you below)

File: src/lib/util.js [hunk 3]
```diff
@@ -508,28 +521,33 @@ export function loadScripts(urls, disableAMD = false) {
...
+        promises.push(promiseMap[url]);
     });
 
     return Promise.all(promises)
         .then(() => {
             if (disableAMD && amdPresent) {
                 define = defineRef;
             }
+
+            clearPromises();
```
Comment (DanDeMicco) [inline src/lib/util.js]: can this go into a `.finally`?

**Pull request closed without merging**